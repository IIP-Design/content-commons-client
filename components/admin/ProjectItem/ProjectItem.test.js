import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProjectItem from './ProjectItem';

const Trigger = () => <div className="trigger">Trigger</div>;
const Content = () => <div className="content">Content</div>;
const props = {
  projectId: '123',
  itemId: '3728',
  isAvailable: false,
  displayItemInModal: true,
  modalTrigger: Trigger,
  modalContent: Content
};

const Component = <ProjectItem { ...props } />;

describe( '<ProjectItem />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders initially the Placeholder component', () => {
    const wrapper = shallow( Component );
    expect( wrapper.name() ).toEqual( 'Placeholder' );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders a Modal component if `isAvailable` and `displayItemInModal`', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { isAvailable: true } );
    expect( wrapper.name() ).toEqual( 'Modal' );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'Modal component receives the Trigger component as prop', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { isAvailable: true } );
    expect( wrapper.prop( 'trigger' ).type() )
      .toEqual( Trigger() );
  } );

  it( 'Modal component receives the Content component as child', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { isAvailable: true } );
    const content = wrapper.find( 'Content' );
    expect( content.type()() ).toEqual( Content() );
  } );

  it( 'renders the Trigger component only if `isAvailable` and `!displayItemInModal`', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( {
      isAvailable: true,
      displayItemInModal: false
    } );
    expect( wrapper.type()() ).toEqual( Trigger() );
    expect( wrapper.props() )
      .toEqual( {
        projectId: '123',
        itemId: '3728',
        displayItemInModal: false
      } );
  } );
} );
