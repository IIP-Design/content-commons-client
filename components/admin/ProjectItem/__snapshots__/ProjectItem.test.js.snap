// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectItem /> renders a Modal component if \`isAvailable\` and \`displayItemInModal\` 1`] = `
<Modal
  centered={true}
  closeIcon={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  trigger={
    <Trigger
      displayItemInModal={true}
      itemId="3728"
      projectId="123"
    />
  }
>
  <ModalContent>
    <Content />
  </ModalContent>
</Modal>
`;

exports[`<ProjectItem /> renders initially the Placeholder component 1`] = `
<Placeholder
  childEl="div"
  childStyles={
    Object {
      "heading": Object {
        "cursor": "not-allowed",
        "height": "1.3em",
        "marginTop": "0.625em",
        "width": "80%",
      },
      "language": Object {
        "cursor": "not-allowed",
        "width": "60%",
      },
      "thumbnail": Object {
        "cursor": "not-allowed",
        "height": "100%",
        "left": "0",
        "position": "absolute",
        "top": "0",
        "width": "100%",
      },
    }
  }
  parentEl="li"
  parentStyles={
    Object {
      "height": "0",
      "marginBottom": "1.5em",
      "paddingTop": "56.25%",
      "position": "relative",
    }
  }
/>
`;
